### PR TITLE
Modify run conditions for nightly job to create test cycle and post in channel.

### PR DESCRIPTION
**Pull request by yasserfaraazkhan**
- e2e is running smoothly along the nightly job (https://github.com/mattermost/desktop/actions/runs/7809764570/job/21302110903). 

However, we need nightly to also create test execution cycles in zephyr.
This PR is to create different Zephyr folders and to enable nighly job to create test cycles.

image

- Link to report in [community](https://community.mattermost.com/core/pl/h9t3z56seby47n5zpgb7erj8ue)

```release-note
NONE
```

### CODE REVIEW

File: e2e/utils/test_cases.js [hunk 1]
```diff
@@ -83,13 +85,47 @@ function saveToEndpoint(url, data) {
...
+            linux: 12358649,
+            default: ZEPHYR_FOLDER_ID,
+        },
+        NIGHTLY: {
+            darwin: 12363687,
+            win32: 12363690,
+            linux: 12363689,
+            default: ZEPHYR_FOLDER_ID,
+        },
+    };
```
Comment (saturninoabril) [inline e2e/utils/test_cases.js:114]: I feel like this should not be hardcoded here, and should be declared depending on OS matrix strategy at workflow file.

File: package.json [hunk 2]
```diff
@@ -164,7 +164,7 @@
     "copy-webpack-plugin": "10.2.4",
     "cross-env": "7.0.3",
     "css-loader": "6.7.1",
-    "electron": "29.0.0",
+    "electron": "29.1.0",
```
Comment (saturninoabril) [inline package.json:167]: Should this be updated here or separately when all dependencies are scheduled for upgrade?

File: .github/workflows/e2e-functional.yml [hunk 3]
```diff
@@ -65,17 +64,16 @@ jobs:
...
       }}
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
     steps:
-
       - name: ci/checkout-repo
         uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
       - name: ci/setup-node
-        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        uses: actions/setup-node@v4
```
Comment (saturninoabril) [inline .github/workflows/e2e-functional.yml:76]: Please follow the same convention by using fix commit hash and version in comment. I believe the main `mattermost` repo is using v4 already.

**Review by saturninoabril — COMMENTED** [review]
Thanks @yasserfaraazkhan! Few clarifications only but overall looks good.

File: .github/workflows/e2e-functional.yml (hunk 3 above)
Comment (yasserfaraazkhan) [reply .github/workflows/e2e-functional.yml:76]: sure Saturn.

File: package.json (hunk 2 above)
Comment (yasserfaraazkhan) [reply package.json:167]: I was seeing few tests break due to white screen. I am check i case the upgrade would fix it. I'll revert this.

File: e2e/utils/test_cases.js (hunk 1 above)
Comment (yasserfaraazkhan) [reply e2e/utils/test_cases.js:114]: @saturninoabril  sure Saturn, that would be a good place to keept this

File: .github/workflows/e2e-functional.yml [hunk 4]
```diff
@@ -89,38 +87,40 @@ jobs:
           wget -qO /usr/local/bin/yq https://github.com/mikefarah/yq/releases/download/v4.20.1/yq_linux_amd64 && chmod a+x /usr/local/bin/yq
           sudo apt-get update || true && sudo apt-get install -y ca-certificates libxtst-dev libpng++-dev gcc-aarch64-linux-gnu g++-aarch64-linux-gnu jq icnsutils graphicsmagick tzdata
           npm ci
+          npm install -g node-gyp
+          npm i robotjs
```
Comment (devinbinnie) [inline .github/workflows/e2e-functional.yml:91]: Why are we installing this and `node-gyp` separately? Can we not just use what's already in the Desktop App? There is a script to do all of this already.

**Review by devinbinnie — CHANGES_REQUESTED** [review]

File: e2e/utils/known_flaky_tests.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    ],
+    "linux": [
+        "menu/view MM-T820 should open Developer Tools For Application Wrapper for main window",
+        "Menu/window_menu MM-T824 should be minimized when keyboard shortcuts are pressed",
+        "Menu/window_menu MM-T825 should be hidden when keyboard shortcuts are pressed",
+        "header MM-T2637 Double-Clicking on the header should minimize/maximize the app MM-T2637_1 should maximize on double-clicking the header"
+    ],
+    "win32": [
+        "application MM-T1304/MM-T1306 should open the app on the requested deep link"
+    ]
```
Comment (yasserfaraazkhan) [inline e2e/utils/known_flaky_tests.json:13]: this is a temp solution to keep a track of flaky tests.

**Review by hmhealey — APPROVED** [review]

**Pull request merged**